cloud_functions/gcs_event_based_ingest/gcs_ocn_bq_ingest/main.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2018 Google LLC.
+# This software is provided as-is, without warranty or representation
+# for any use or purpose.
+# Your use of it is subject to your agreement with Google.
+
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Background Cloud Function for loading data to BigQuery.
+"""
+import json
+from collections import deque
+from os import getenv
+from pathlib import Path
+from typing import Any
+from typing import Dict, List, Optional, Tuple
+from urllib.parse import urlparse
+import logging
+from time import monotonic, sleep
+
+from google.api_core.client_info import ClientInfo
+from google.cloud import bigquery, storage
+from google.cloud.exceptions import NotFound
+
+# 15TB per BQ load job.
+MAX_BATCH_BYTES = 15 * 10 ** 12
```

Comment:
this should be configurable w/ environment variable for customers who have negotiated a higher per load job bytes limit.